Pully Penelope/Assets/Scripts/PlayerMovement.cs

```diff
@@ -36,21 +70,128 @@ private void FixedUpdate()
-            animator.SetFloat("moveY", positionChange.y);
+            if (!isGrabbingHandle)
+            {
+                animator.SetFloat("moveX", positionChange.x);
+                animator.SetFloat("moveY", positionChange.y);
+            }
+            /*else if (isGrabbingHandle && shouldUpdateAnimation)
+            {
+                Debug.Log("Work");
+                animator.SetFloat("moveX", handleOrientation.x);
+                animator.SetFloat("moveY", handleOrientation.y);
+                //shouldUpdateAnimation = false;
+            }*/
             animator.SetBool("isMoving", true);
         }
         else
         {
+            if (isGrabbingHandle && shouldUpdateAnimation)
+            {
+                animator.SetFloat("moveX", handleOrientation.x);
+                animator.SetFloat("moveY", handleOrientation.y);
+                shouldUpdateAnimation = false;
+            }
             animator.SetBool("isMoving", false);
         }
+        CheckIfGrabbing();
+        /*if (handleName == "DHandle")
+        {
+            handleOrientation.x = 0;
+            handleOrientation.y = 1;
+        }
+        else if (handleName == "UHandle")
+        {
+            handleOrientation.x = 0;
+            handleOrientation.y = -1;
+        }
+        else if (handleName == "RHandle")
+        {
+            handleOrientation.x = -1;
+            handleOrientation.y = 0;
+        }
+        else if (handleName == "LHandle")
+        {
+            handleOrientation.x = -1;
+            handleOrientation.y = 0;
+        }*/
+        //if (handleTransform != null)
+        //{
+        //Vector2 D = transform.position - handleTransform.position; // line from crate to player
+        //float dist = D.magnitude;
+        //Vector2 pullDir = D.normalized; // short blue arrow from crate to player
+        //if (dist > 50) handleTransform = null; // lose tracking if too far
+        //else if (dist > 3)
+        //{ don't pull if too close
+        // this is the same math to apply fake gravity. 10 = normal gravity
+        //float pullF = 10;
+        // for fun, pull a little bit more if further away:
+        // (so, random, optional junk):
+        //float pullForDist = (dist - 3) / 2.0f;
+        //if (pullForDist > 20) pullForDist = 20;
+        //pullF += pullForDist;
+        // Now apply to pull force, using standard meters/sec converted
+        //    into meters/frame:
+        //handleTransform.GetComponent<Rigidbody2D>().velocity += pullDir * (pullF * Time.deltaTime);
+        //}
+        //}
     }
 
     private void MovePlayer()
     {
         rigidbody.MovePosition(transform.position + positionChange.normalized * moveSpeed * Time.deltaTime);
+        if (handleName != null && grabFloat != 0)
+        {
+            handleObject.GetComponent<Rigidbody2D>().MovePosition(handleTransform.position + positionChange.normalized * moveSpeed * Time.deltaTime);
+        }
+    }
+
+    private void CheckIfGrabbing()
```

Comment:
IsGrabbing would make sense to turn into a property, and you can set the animator bools in the property